python/pyarrow/parquet.py

```diff
@@ -871,21 +889,34 @@ def filter_accepts_partition(self, part_key, filter, level):
         if p_column != f_column:
             return True
 
-        f_type = type(f_value)
+        p_value = self.levels[level].dictionary[p_value_index].as_py()
 
-        if isinstance(f_value, set):
+        if op in {'in', 'not in'}:
+            if not (isinstance(f_value, Container) and
+                    isinstance(f_value, Iterable)):
```

Comment:
Can you use the abstract types from `collections.abc` ?  
Also, we need to ensure we don't accidentally allow a single string value this way (`("part", "in", "abc")` would then be interpreted like `("part", "in", ["a", "b", "c"])`, which I don't think we should do)

Reply:
Now ('part', 'in', 'abc') will be True for 'a', 'b', 'ab' or 'abc'. There is a new test case.